cmd/experimental/kjobctl/pkg/builder/slurm_builder_test.go

```diff
@@ -233,57 +233,75 @@ func TestSlurmBuilderDo(t *testing.T) {
 				CompletionMode(batchv1.IndexedCompletion).
 				Profile("profile").
 				Mode(v1alpha1.SlurmMode).
+				WithInitContainer(*wrappers.MakeContainer("slurm-init-env", "bash:5-alpine3.20").
+					Command("bash", "/slurm/scripts/init-entrypoint.sh").
+					WithVolumeMount(corev1.VolumeMount{Name: "slurm-scripts", MountPath: "/slurm/scripts"}).
+					WithVolumeMount(corev1.VolumeMount{Name: "slurm-env", MountPath: "/slurm/env"}).
+					Obj()).
 				WithContainer(*wrappers.MakeContainer("c1", "bash:4.4").
-					Command("bash", "/slurm/entrypoint.sh").
-					WithVolumeMount(corev1.VolumeMount{MountPath: "/slurm"}).
+					Command("bash", "/slurm/scripts/entrypoint.sh").
+					WithVolumeMount(corev1.VolumeMount{Name: "slurm-scripts", MountPath: "/slurm/scripts"}).
+					WithVolumeMount(corev1.VolumeMount{Name: "slurm-env", MountPath: "/slurm/env"}).
+					WithEnvVar(corev1.EnvVar{Name: constants.EnvVarNameUserID, Value: userID}).
+					WithEnvVar(corev1.EnvVar{Name: constants.EnvVarTaskName, Value: "default_profile"}).
+					WithEnvVar(corev1.EnvVar{
+						Name:  constants.EnvVarTaskID,
+						Value: fmt.Sprintf("%s_%s_default_profile", userID, testStartTime.Format(time.RFC3339)),
+					}).
+					WithEnvVar(corev1.EnvVar{Name: "PROFILE", Value: "default_profile"}).
+					WithEnvVar(corev1.EnvVar{Name: "TIMESTAMP", Value: testStartTime.Format(time.RFC3339)}).
+					WithEnvVar(corev1.EnvVar{Name: "JOB_CONTAINER_INDEX", Value: "0"}).
 					Obj()).
 				WithVolume(corev1.Volume{
+					Name: "slurm-scripts",
 					VolumeSource: corev1.VolumeSource{
 						ConfigMap: &corev1.ConfigMapVolumeSource{
 							Items: []corev1.KeyToPath{
+								{Key: "init-entrypoint.sh", Path: "init-entrypoint.sh"},
 								{Key: "entrypoint.sh", Path: "entrypoint.sh"},
 								{Key: "script", Path: "script", Mode: ptr.To[int32](0755)},
 							},
 						},
 					},
 				}).
-				WithEnvVar(corev1.EnvVar{Name: constants.EnvVarNameUserID, Value: userID}).
-				WithEnvVar(corev1.EnvVar{Name: constants.EnvVarTaskName, Value: "default_profile"}).
-				WithEnvVar(corev1.EnvVar{
-					Name:  constants.EnvVarTaskID,
-					Value: fmt.Sprintf("%s_%s_default_profile", userID, testStartTime.Format(time.RFC3339)),
+				WithVolume(corev1.Volume{
+					Name: "slurm-env",
+					VolumeSource: corev1.VolumeSource{
+						EmptyDir: &corev1.EmptyDirVolumeSource{},
+					},
```

Comment:
```suggestion
					VolumeSource: corev1.VolumeSource{},
```

Couldn't we use the `cmpopts.EquateEnpty()` instead of such an empty struct?

Reply:
Is it working with pointers in struct?

```
          						VolumeSource: v1.VolumeSource{
          							HostPath:             nil,
        - 							EmptyDir:             nil,
        + 							EmptyDir:             s"&EmptyDirVolumeSource{Medium:,SizeLimit:<nil>,}",
          							GCEPersistentDisk:    nil,
          							AWSElasticBlockStore: nil,
          							... // 26 identical fields
          						},
```

Reply:
Looks like only only slices and maps: https://github.com/google/go-cmp/blob/c3ad8435e7bef96af35732bc0789e5a2278c6d5f/cmp/cmpopts/equate.go#L20-L26

Reply:
That makes sense. Thank you for investigating this.